### DISCUSSION TITLE
String sort change in .NET 5.0

### DISCUSSION THREAD

**Discussion by RussKie** [dotnet/runtime#46265] [Q&A] [answered]
Running the following sample yields different results in .NET 5.0 compared to .NET Framework and .NET Core:
```cs
List<string> list = new List<string>() { "current/2", "current_2" };
list.Sort();

foreach (string s in list)
{
	Console.WriteLine(s);
}
```

.NET Framework / .NET Core:
```
current/2
current_2
```

.NET 5.0:
```
current_2
current/2
```

@GrabYourPitchforks is this known? I've searched through the repo but couldn't find anything obvious.

**Comment by gfoidl** [comment, marked as answer]
See [Globalization APIs use ICU libraries on Windows](https://docs.microsoft.com/en-us/dotnet/core/compatibility/globalization/5.0/icu-globalization-api) and the issues here regarding _ICU_.

**Reply by RussKie** [reply]
Thank you, I kind of suspected it may be related.

**Comment by GrabYourPitchforks** [comment]
@gfoidl's answer is correct. By default, `string.CompareTo` (used by `List<T>.Sort`) defaults to the linguistic `StringComparer.CurrentCulture`. I don't know much about your scenario, but I suspect you probably wanted a non-linguistic comparer. Try calling `List<T>.Sort(IComparer<T>)` and pass `StringComparer.Ordinal` or `StringComparer.OrdinalIgnoreCase` if you want non-linguistic behavior.

(Note: using `StringComparer`, not the `StringComparison` enum.)

**Reply by RussKie** [reply]
Thank you
